### PR TITLE
postgres 15 support

### PR DESCRIPTION
**Pull request by RekGRpth**
postgres 15 support

this fixes https://github.com/postgrespro/pg_wait_sampling/issues/53

### CODE REVIEW

**Comment by maksm90** [comment]
This change was induced by recent change in commit https://github.com/postgres/postgres/commit/ce3049b0215b63744d11c0ce3ac6afdb67fc2ff0 . 

Generally LGTM. But @RekGRpth could you take multiple calls of `InitPostgres()` away to `compat.c` where all these calls are hidden under unified name with `_compat` suffix.

**Comment by RekGRpth** [comment]
> This change was induced by recent change in commit [postgres/postgres@ce3049b](https://github.com/postgres/postgres/commit/ce3049b0215b63744d11c0ce3ac6afdb67fc2ff0) .

No, but this https://github.com/postgres/postgres/commit/31ed3cf746a61538876c66a778ec8a372a3b4ffb

**Comment by RekGRpth** [comment]
> could you take multiple calls of `InitPostgres()` away to `compat.c` where all these calls are hidden under unified name with `_compat` suffix.

done

File: compat.c [hunk 1]
```diff
@@ -33,3 +33,18 @@ CreateTemplateTupleDescCompat(int nattrs, bool hasoid)
 	return CreateTemplateTupleDesc(nattrs, hasoid);
 #endif
 }
+
+inline void InitPostgresCompat(const char *in_dbname, Oid dboid,
```
Comment (maksm90) [inline compat.c:37]: Please, follow here BSD style of code formatting - function type has to be located on preceded line from its name definition.

**Review by maksm90 — APPROVED** [review]

**Comment by maksm90** [comment]
> No, but this https://github.com/postgres/postgres/commit/31ed3cf746a61538876c66a778ec8a372a3b4ffb

Yes, I had erroneously copy-pasted the link.

@shinderuk after fixing of my note PR might be merged.

File: compat.c (hunk 1 above)
Comment (RekGRpth) [reply compat.c]: done

**Comment by shinderuk** [comment]
Merged after squashing and adding a missing include. Thank you!

**Comment by df7cb** [comment]
Hi,
PG 15.0 is due in two weeks, could you please tag a new pg_wait_sampling release so we can update the packages?
Thanks!

**Comment by maksm90** [comment]
> Hi,
PG 15.0 is due in two weeks, could you please tag a new pg_wait_sampling release so we can update the packages?

Hi, sure! You might track the progress in https://github.com/postgrespro/pg_wait_sampling/issues/59
Thanks for notice.

**Pull request merged**